### DISCUSSION TITLE
cannot get version information: commit not found

### DISCUSSION THREAD

**Discussion by mcoz** [qdm12/gluetun#684] [Q&A] [answered]
I am seeing the following error message using both docker run and docker-compose using the examples provided:

`ERROR vpn: cannot get version information: commit not found: 3651cc6161f390665b09dc7b213ed742eb2d9d2e`

I am using Ubuntu 20.04 and have found the error on two different servers. Also getting the same error using containers for two different VPN providers - PIA and Nordvpn.

The container is connecting to the VPN and other containers can connect through it.  The terminal is not however returning to a command prompt and I need to open a second terminal window to interact with the container.

I suspect I haven't provided enough technical detail so please let me know what else I can provide.

Thanks in advance

**Comment by mcoz** [comment]
Never mind - adding a version tag fixed the error. 

I'm very much a docker pre-schooler

**Reply by Sea-Wolfe** [reply]
For clarity:
Adding a version tag for the docker image?
Is that what fixed the problem?
Tx

**Reply by mcoz** [reply]
Yes - pulling the image again fixed the error - the rest of my message referenced my not understanding the default option for the docker-compose up command.

**Comment by qdm12** [comment]
You might had pulled a bad commit I reverted (I was experimenting with CI a few days ago and overriding the last commit a few times). That commit doesn't exist indeed https://github.com/qdm12/gluetun/search?q=3651cc6161f390665b09dc7b213ed742eb2d9d2e&type=

What's the build date mentioned at the top of the logs?

Try pulling the image again, it should be working fine. Note it's just a warning, it won't affect anything except you not knowing if there is a new release etc.

**Reply by Sea-Wolfe** [reply]
Thanks! (I was also seeing the same message if my logs).  So I'll try pulling the image again.
Great work on this, btw!
I tried a couple of the other similar openvpn docker images, and while I did get one of the others to work, yours was by far the easiest to get working-great documentation on wiki etc.
I'm using a custom provider, and I was thinking about submitting an issue/request to see if you can add it. (Since you are responding here, I'll ask before doing that):
Would having that provider added make it any easier to change/select servers vs manually editing the custom.ovpn file?  Or is that the only way?

**Comment by Sea-Wolfe** [comment]
For the benefit of anyone else searching/reading:

Even after deleting the image, and pulling with tag "latest" this message persists.  For me, I changed to "image: qmcgaw/gluetun:**master** and that pulled version 3.25 and this message went away (replaced by this message):

" INFO vpn: There is a new release v3.25.0 (v3.25.0) created 22 days ago"

**Reply by qdm12** [reply]
Actually that `:master` image was a mistake, it's deleted now 😄 And `:latest` is fixed now (see comment below).

**Comment by qdm12** [comment, marked as answer]
It's fixed now. The last commit 3651cc6161f390665b09dc7b213ed742eb2d9d2e 5 days ago changed the CI build to use long commits instead of short commits (7 characters). I changed that back to use short commits and everything works again. Thanks for letting me know!

**Reply by Sea-Wolfe** [reply]
Thanks!  I will pull latest in a bit, and test again.

In the meantime do you mind responding about whether I should request to add another VPN provider?  As is, everything is working fine, so I was just wondering if having the provider added, adds any additional funtionality/configuration options.  I don't want to waste time (yours and mine), opening an issue, if it's not going to add anything to the way it's running right now.

Thanks again for your awesome work on this!  Great job!

**Reply by qdm12** [reply]
> Would having that provider added make it any easier to change/select servers vs manually editing the custom.ovpn file? Or is that the only way?

It depends. Some providers such as Cyberghost are a PITA even when 'natively supported', so sometimes it's even easier to just use the 'custom provider'. Now some others are much easier to use. You can just set your username, password and pick a country/city/other filter through env variables, so it's more fun to switch. One day (I hope!), they will also be mutable through a mobile app, whilst custom config will be way less fun to modify 😄.

What provider is it? Feel free to send me an example openvpn config file so I can let you know if it's worth it or not. e.g. if they attach some sort of private key UNIQUE by client AND by server, there is little point since you would have to bind mount 2 files instead of 1 whenever you want to change location.

**Reply by Sea-Wolfe** [reply]
Got it!
Well, before finding your image, this was my attempt at a custom config using another image here:

https://github.com/Sea-Wolfe/vpn-configs-contrib/commit/e8ff78b44db655ef340c6df14cd46a8933f56169

It was my first time trying to fork anything or use GitHub in anyway, other than searching for stuff..lol.
The provider is VPNSecure.me  --I don't think it's one of of the big popular ones.  But I bought a "lifetime" sub to it sometime back for super cheap, and it works for my needs--not a heavy user. 

If the ovpn example above is not clear (I was messing around--multiple tries--so I might have messed up), I can email you all or files, so you can take a look, or share them via some other method (open to suggestions).  I am not in an super hurry and this is not some pressing issue, so I will work with you at your speed.

Thank you for your time and attention!

**Reply by qdm12** [reply]
Yes please share all of them except your username+password, my email is quentin.mcgaw@gmail.com.

It looks like they email a zip file, so no way to really know if some of the values (like some key files) are user specific or not. I'll just pretend they're not until someone else than you complains it doesn't work 😄

I created #685 for it, feel free to subscribe to it to be alerted. We can also continue the conversation there.
